Compilation Warning

docker run --mount type=bind,source=.,target=/TicTacToe --rm tic-tac-toe make -f linux.mk
mkdir -p bin/build_linux//TicTacToe/GameBoard/
g++  -std=c++17 -Wall -Wextra -g -Werror -static-libgcc -static-libstdc++ -I/TicTacToe -I/TicTacToe/.github -I/TicTacToe/.github/workflows -I/TicTacToe/GameBoard -I/TicTacToe/Arena -I/TicTacToe/Players -I/TicTacToe/Logger -I/TicTacToe/DrawTools -I/TicTacToe/.git -I/TicTacToe/.git/logs -I/TicTacToe/.git/logs/refs -I/TicTacToe/.git/logs/refs/heads -I/TicTacToe/.git/logs/refs/remotes -I/TicTacToe/.git/logs/refs/remotes/origin -I/TicTacToe/.git/hooks -I/TicTacToe/.git/objects -I/TicTacToe/.git/objects/ab -I/TicTacToe/.git/objects/0b -I/TicTacToe/.git/objects/47 -I/TicTacToe/.git/objects/60 -I/TicTacToe/.git/objects/cb -I/TicTacToe/.git/objects/e4 -I/TicTacToe/.git/objects/d7 -I/TicTacToe/.git/objects/d1 -I/TicTacToe/.git/objects/0d -I/TicTacToe/.git/objects/bc -I/TicTacToe/.git/objects/4d -I/TicTacToe/.git/objects/bf -I/TicTacToe/.git/objects/69 -I/TicTacToe/.git/objects/28 -I/TicTacToe/.git/objects/b8 -I/TicTacToe/.git/objects/d5 -I/TicTacToe/.git/objects/8d -I/TicTacToe/.git/objects/a9 -I/TicTacToe/.git/objects/a2 -I/TicTacToe/.git/objects/96 -I/TicTacToe/.git/objects/06 -I/TicTacToe/.git/objects/ee -I/TicTacToe/.git/objects/info -I/TicTacToe/.git/objects/79 -I/TicTacToe/.git/objects/e8 -I/TicTacToe/.git/objects/25 -I/TicTacToe/.git/objects/fd -I/TicTacToe/.git/objects/cf -I/TicTacToe/.git/objects/ff -I/TicTacToe/.git/objects/e6 -I/TicTacToe/.git/objects/e3 -I/TicTacToe/.git/objects/5c -I/TicTacToe/.git/objects/2d -I/TicTacToe/.git/objects/bb -I/TicTacToe/.git/objects/61 -I/TicTacToe/.git/objects/pack -I/TicTacToe/.git/objects/72 -I/TicTacToe/.git/objects/a0 -I/TicTacToe/.git/objects/ac -I/TicTacToe/.git/objects/91 -I/TicTacToe/.git/info -I/TicTacToe/.git/branches -I/TicTacToe/.git/refs -I/TicTacToe/.git/refs/heads -I/TicTacToe/.git/refs/remotes -I/TicTacToe/.git/refs/remotes/origin -I/TicTacToe/.git/refs/tags -I/TicTacToe/Commons -MMD -MP -O3  -c /TicTacToe/GameBoard/GameBoard.cpp -o bin/build_linux//TicTacToe/GameBoard/GameBoard.cpp.o
/TicTacToe/GameBoard/GameBoard.cpp: In member function 'bool GameBoard::IsEndOfGame()':
/TicTacToe/GameBoard/GameBoard.cpp:95:43: error: left operand of comma operator has no effect [-Werror=unused-value]
   95 |     for(int i = 0, k = iBoardSize - 1; (i < iBoardSize), (k >= 0); i++, k--)
      |                                        ~~~^~~~~~~~~~~~~
/TicTacToe/GameBoard/GameBoard.cpp: In member function 'bool GameBoard::SaveVictorySet(const Commons::MatrixProperties&, const int&)':
/TicTacToe/GameBoard/GameBoard.cpp:256:64: error: left operand of comma operator has no effect [-Werror=unused-value]
  256 |         for(int iCount = 0, jCount = (iBoardSize - 1); (iCount < iBoardSize), (jCount >= 0); iCount++, jCount--)
      |                                                        ~~~~~~~~^~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make: *** [linux.mk:27: bin/build_linux//TicTacToe/GameBoard/GameBoard.cpp.o] Error 1
make: *** [makefile:7: linux] Error 2
Error: Process completed with exit code 2.

--- a/.github/workflows/master-workflow.yml
+++ b/.github/workflows/master-workflow.yml
@@ -5,11 +5,15 @@ on:
     branches: [ master ]
 
 
-
 jobs:
   create_release:
   
     runs-on: ubuntu-latest
+    container: 
+      image: ghcr.io/nofea/tic-tac-toe:latest
+      credentials:
+        username: nofea
+        password: ${{secrets.REGISTRY_PASS}}
 
     steps:
     - name: Install dependencies
@@ -62,7 +66,7 @@ jobs:
       id: create_release
       uses: actions/create-release@v1
       env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # This token is provided by Actions, you do not need to create your own token
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       with:
         tag_name: ${{ github.ref }}
         release_name: Release ${{ github.ref }}

--- a/.github/workflows/pr-workflow.yml
+++ b/.github/workflows/pr-workflow.yml
@@ -10,6 +10,11 @@ jobs:
   build:
   
     runs-on: ubuntu-latest
+    container: 
+      image: ghcr.io/nofea/tic-tac-toe:latest
+      credentials:
+        username: nofea
+        password: ${{secrets.REGISTRY_PASS}}
 
     steps:
       - name: Install dependencies

--- a/.github/workflows/pr-workflow.yml
+++ b/.github/workflows/pr-workflow.yml
@@ -5,7 +5,6 @@ on:
     branches: [ master ]
 
 
-
 jobs:
   build:
   

--- a/Dockerfile
+++ b/Dockerfile
@@ -4,6 +4,6 @@ WORKDIR /TicTacToe
 
 COPY . .
 
-RUN apt -y update
+RUN apt -y update && apt -y upgrade
 
 RUN apt -y install $(cat requirements.txt)

--- a/GameBoard/GameBoard.cpp
+++ b/GameBoard/GameBoard.cpp
@@ -92,7 +92,7 @@ bool GameBoard::IsEndOfGame()
     int iTemp = GAMEBOARDINITVALUE;
     list<int> listRowX, listRowO, listColX, listColO, listPDiaX, listPDiaO, listSDiaX, listSDiaO;   
     
-    for(int i = 0, k = iBoardSize - 1; (i < iBoardSize), (k >= 0); i++, k--)
+    for(int i = 0, k = iBoardSize - 1; (i < iBoardSize) && (k >= 0); i++, k--)
     {
         //check the rows and cols
         for(int j = 0; j < iBoardSize; j++)
@@ -253,7 +253,7 @@ bool GameBoard::SaveVictorySet(const Commons::MatrixProperties& Property, const 
         break;
 
         case objCommons->MatrixProperties_SecondaryDiagonal:
-        for(int iCount = 0, jCount = (iBoardSize - 1); (iCount < iBoardSize), (jCount >= 0); iCount++, jCount--)
+        for(int iCount = 0, jCount = (iBoardSize - 1); (iCount < iBoardSize) && (jCount >= 0); iCount++, jCount--)
         {
             listVictorySet.push_back(pair<int,int> {iCount, jCount});    
         }

--- a/Players/ComputerBlocker.cpp
+++ b/Players/ComputerBlocker.cpp
@@ -188,7 +188,7 @@ bool ComputerBlocker::MakeVictoryConditionsList(int& iGameBoardSize)
 {
     list<pair<int,int>> listTempRow, listTempCol, listTempPDia, listTempSDia;
 
-    for(int j = 0, i = (iGameBoardSize - 1); (j < iGameBoardSize), (i >= 0); j++, i--)
+    for(int j = 0, i = (iGameBoardSize - 1); (j < iGameBoardSize) && (i >= 0); j++, i--)
     {
         for (int k = 0; k < iGameBoardSize; k++)
         {

--- a/config.mk
+++ b/config.mk
@@ -9,7 +9,7 @@ DOXYGEN_DIR:=Documentation
 
 LINT_DIR:=lintresults
 
-DOCKER_CONTAINER:=tic-tac-toe
+DOCKER_CONTAINER:=ghcr.io/nofea/tic-tac-toe
 
 TARGET_EXEC_LINUX?=tictactoe.out
 

--- a/makefile
+++ b/makefile
@@ -1,19 +1,39 @@
 include config.mk
 
 create_docker_image:
+ifeq (, $(shell which docker))
+	$(error "No docker installed")
+else
 	docker build -t $(DOCKER_CONTAINER) .
+endif
 
 linux:
+ifeq (, $(shell which docker))
+	$(MAKE) -f linux.mk
+else
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(MAKE) -f linux.mk
+endif
 
 windows:
+ifeq (, $(shell which docker))
+	$(MAKE) -f windows.mk
+else
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(MAKE) -f windows.mk
+endif
 
 lint:
+ifeq (, $(shell which docker))
+	$(MAKE) -f lint.mk
+else
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(MAKE) -f lint.mk
+endif
 
 doc:
+ifeq (, $(shell which docker))
+	$(DOXYGEN) ttt_doxy
+else
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(DOXYGEN) ttt_doxy
+endif
 
 run_linux:
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(BUILD_DIR_LINUX)/$(TARGET_EXEC_LINUX)
@@ -23,16 +43,32 @@ all: linux windows
 .PHONY: clean
 
 clean_container:
+ifeq (, $(shell which docker))
+	$(error "No docker installed")
+else
 	docker image rm -f $(DOCKER_CONTAINER)
+endif
 
 clean:
+ifeq (, $(shell which docker))
+	$(RM) -r $(BUILD_DIR)
+else
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(RM) -r $(BUILD_DIR)
+endif
 
 clean_lint:
+ifeq (, $(shell which docker))
+	$(RM) -r $(LINT_DIR)
+else
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(RM) -r $(LINT_DIR)
+endif
 
 clean_doc:
+ifeq (, $(shell which docker))
+	$(RM) -r $(DOXYGEN_DIR)
+else
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(RM) -r $(DOXYGEN_DIR)
+endif
 
 very_clean:clean clean_lint clean_doc clean_container
 

--- a/makefile
+++ b/makefile
@@ -3,19 +3,19 @@ include config.mk
 create_docker_image:
 	docker build -t $(DOCKER_CONTAINER) .
 
-linux: create_docker_image
+linux:
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(MAKE) -f linux.mk
 
-windows: create_docker_image
+windows:
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(MAKE) -f windows.mk
 
-lint: create_docker_image
+lint:
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(MAKE) -f lint.mk
 
-doc: create_docker_image
+doc:
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(DOXYGEN) ttt_doxy
 
-run_linux: create_docker_image
+run_linux:
 	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(BUILD_DIR_LINUX)/$(TARGET_EXEC_LINUX)
 
 all: linux windows
@@ -26,13 +26,13 @@ clean_container:
 	docker image rm -f $(DOCKER_CONTAINER)
 
 clean:
-	$(RM) -r $(BUILD_DIR)
+	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(RM) -r $(BUILD_DIR)
 
 clean_lint:
-	$(RM) -r $(LINT_DIR)
+	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(RM) -r $(LINT_DIR)
 
 clean_doc:
-	$(RM) -r $(DOXYGEN_DIR)
+	docker run --mount type=bind,source=.,target=/TicTacToe --rm $(DOCKER_CONTAINER) $(RM) -r $(DOXYGEN_DIR)
 
 very_clean:clean clean_lint clean_doc clean_container
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,1 +1,1 @@
-apt-utils libc6 g++ doxygen gcc-mingw-w64 mingw-w64 cppcheck make
+apt-utils g++ doxygen gcc-mingw-w64 mingw-w64 cppcheck make

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,1 +1,1 @@
-g++ doxygen gcc-mingw-w64 mingw-w64 cppcheck make
+apt-utils libc6 g++ doxygen gcc-mingw-w64 mingw-w64 cppcheck make

